### PR TITLE
Add proto message, implement sdk.Msg for Recover Client.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

closes: #4493

this only takes care of adding proto and then satisfying the compiler, follow ups will implement func for client keeper and on msg_server.

### Commit Message / Changelog Entry

```bash
feat: add proto message for RecoverClient
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by crodriguezvega — APPROVED** [review]
Looks very good!

File: modules/core/02-client/types/msgs_test.go [hunk 1]
```diff
@@ -678,3 +680,87 @@ func TestMsgUpdateParamsGetSigners(t *testing.T) {
 		}
 	}
 }
+
+// TestMsgRecoverClientValidateBasic tests ValidateBasic for MsgRecoverClient
+func (suite *TypesTestSuite) TestMsgRecoverClientValidateBasic() {
+	var msg *types.MsgRecoverClient
+
+	signer := ibctesting.TestAccAddress
```
Comment (damiannolan) [inline modules/core/02-client/types/msgs_test.go:688]: I think this var can be removed, right?

File: modules/core/02-client/types/msgs.go [hunk 2]
```diff
@@ -283,3 +284,32 @@ func (msg *MsgUpdateParams) ValidateBasic() error {
...
+
+// ValidateBasic performs basic checks on a MsgRecoverClient.
+func (msg *MsgRecoverClient) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Signer); err != nil {
+		return errorsmod.Wrapf(ibcerrors.ErrInvalidAddress, "string could not be parsed as address: %v", err)
+	}
+	if err := host.ClientIdentifierValidator(msg.SubjectClientId); err != nil {
+		return err
+	}
+	return host.ClientIdentifierValidator(msg.SubstituteClientId)
```
Comment (damiannolan) [inline modules/core/02-client/types/msgs.go:314]: nit: adding some breathing room :)  

```suggestion
	if _, err := sdk.AccAddressFromBech32(msg.Signer); err != nil {
		return errorsmod.Wrapf(ibcerrors.ErrInvalidAddress, "string could not be parsed as address: %v", err)
	}
	
	if err := host.ClientIdentifierValidator(msg.SubjectClientId); err != nil {
		return err
	}
	
	return host.ClientIdentifierValidator(msg.SubstituteClientId)
```

File: proto/ibc/core/client/v1/tx.proto [hunk 3]
```diff
@@ -127,4 +130,22 @@ message MsgUpdateParams {
 }
 
 // MsgUpdateParamsResponse defines the MsgUpdateParams response type.
-message MsgUpdateParamsResponse {}
\ No newline at end of file
+message MsgUpdateParamsResponse {}
+
+// MsgRecoverClient defines the sdk.Msg type to recover a client.
+message MsgRecoverClient {
+  option (gogoproto.equal)           = false;
```
Comment (charleenfei) [inline proto/ibc/core/client/v1/tx.proto:137]: dont these fields default to `false`?

**Review by damiannolan — APPROVED** [review]
Nice work! Just left some nits :)

File: proto/ibc/core/client/v1/tx.proto [hunk 4]
```diff
@@ -127,4 +130,22 @@ message MsgUpdateParams {
...
+  option (gogoproto.goproto_getters) = false;
+  option (cosmos.msg.v1.signer)      = "signer";
+
+  // signer address (it may be the the address that controls the module, which defaults to x/gov unless overwritten).
+  string signer = 1;
+  // the client identifier for the client to be updated if the proposal passes
+  string subject_client_id = 2;
+  // the substitute client identifier for the client standing in for the subject
```
Comment (charleenfei) [inline proto/ibc/core/client/v1/tx.proto:145]:
```suggestion
  // the substitute client identifier for the client which will replace the subject
```

File: proto/ibc/core/client/v1/tx.proto [hunk 5]
```diff
@@ -127,4 +130,22 @@ message MsgUpdateParams {
...
+message MsgUpdateParamsResponse {}
+
+// MsgRecoverClient defines the sdk.Msg type to recover a client.
+message MsgRecoverClient {
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+  option (cosmos.msg.v1.signer)      = "signer";
+
+  // signer address (it may be the the address that controls the module, which defaults to x/gov unless overwritten).
+  string signer = 1;
```
Comment (charleenfei) [inline proto/ibc/core/client/v1/tx.proto:142]: nit but usually the `signer` field is last on our other Msg types

File: proto/ibc/core/client/v1/tx.proto [hunk 6]
```diff
@@ -27,6 +27,9 @@ service Msg {
 
   // UpdateClientParams defines a rpc handler method for MsgUpdateParams.
   rpc UpdateClientParams(MsgUpdateParams) returns (MsgUpdateParamsResponse);
+
+  // Recover defines a rpc handler method for MsgRecoverClient.
```
Comment (charleenfei) [inline proto/ibc/core/client/v1/tx.proto:31]:
```suggestion
  // RecoverClient defines a rpc handler method for MsgRecoverClient.
```

File: modules/core/keeper/msg_server.go [hunk 7]
```diff
@@ -100,6 +100,11 @@ func (k Keeper) SubmitMisbehaviour(goCtx context.Context, msg *clienttypes.MsgSu
 	return &clienttypes.MsgSubmitMisbehaviourResponse{}, nil
 }
 
+// RecoverClient defines a rpc handler method for MsgRecoverClient.
+func (k Keeper) RecoverClient(goCtx context.Context, msg *clienttypes.MsgRecoverClient) (*clienttypes.MsgRecoverClientResponse, error) { //nolint:revive
```
Comment (charleenfei) [inline modules/core/keeper/msg_server.go:104]:
```suggestion
func (Keeper) RecoverClient(goCtx context.Context, msg *clienttypes.MsgRecoverClient) (*clienttypes.MsgRecoverClientResponse, error) {
```

**Review by charleenfei — APPROVED** [review]
a few small comments but otherwise lgtm!

File: modules/core/02-client/types/msgs_test.go (hunk 1 above)
Comment (DimitrisJim) [reply modules/core/02-client/types/msgs_test.go:688]: can be moved inline during struct initialization, yea

File: proto/ibc/core/client/v1/tx.proto (hunk 3 above)
Comment (DimitrisJim) [reply proto/ibc/core/client/v1/tx.proto:137]: yea yea, this can be dropped.

File: modules/core/keeper/msg_server.go (hunk 7 above)
Comment (DimitrisJim) [reply modules/core/keeper/msg_server.go:104]: yea, I'll remove the nolint directive when adding the implementation for RecoverClient

File: proto/ibc/core/client/v1/tx.proto (hunk 5 above)
Comment (DimitrisJim) [reply proto/ibc/core/client/v1/tx.proto:141]: #4510

**Pull request merged**